storage/remote/write.go

```diff
@@ -122,6 +105,11 @@ func (rws *WriteStorage) ApplyConfig(conf *config.Config) error {
 			return err
 		}
 
+		// Don't allow duplicate remote write configs.
+		if _, ok := newQueues[hash]; ok {
+			return fmt.Errorf("duplicate remote write configs are not allowed, found duplicate for URL: %s", rwConf.URL)
```

Comment:
It would mean that 2 configurations without name that differ by (at least) one secret field would trigger this error, correct? It might not be obvious for the end-user why it happens.

Reply:
This behavior actually doesn't take the secret field value into account as it is using the hash. It also isn't new behavior, just a reordering of logic, I moved it to the beginning as a short circuit to doing any extra work.